app/screens/select_server/select_server.test.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+
+import SelectServer from './select_server.js';
+import {shallowWithIntl} from 'test/intl-test-helper';
+
+describe('SelectServer', () => {
+    const actions = {
+        getPing: jest.fn(),
+        handleServerUrlChanged: jest.fn(),
+        scheduleExpiredNotification: jest.fn(),
+        loadConfigAndLicense: jest.fn(),
+        login: jest.fn(),
+        resetPing: jest.fn(),
+        setLastUpgradeCheck: jest.fn(),
+        setServerVersion: jest.fn(),
+    };
+
+    const baseProps = {
+        actions,
+        hasConfigAndLicense: true,
+        serverUrl: '',
+    };
+
+    test('should match snapshot empty URL string', async () => {
+        const wrapper = shallowWithIntl(
+            <SelectServer {...baseProps}/>,
+        );
+        wrapper.instance().handleConnect();
+        expect(wrapper.state().error).not.toBeNull();
+        expect(wrapper.state().error.intl.id).toMatch('mobile.server_url.empty');
+        expect(wrapper.getElement()).toMatchSnapshot();
+    });
+});
```

Comment:
May not be related to the ticket requirement but for sanity check, it will be better to add more tests - one with another error type and the other with happy path, like:
```javascript
    test('should match error when URL does not start with http:// or https://', async () => {
        const {getByTestId, getByText} = renderWithReduxIntl(
            <SelectServer {...baseProps}/>,
        );

        const urlInput = getByTestId('server_url_input');
        fireEvent.changeText(urlInput, 'ht://invalid:8065');

        const button = getByText('Connect');
        fireEvent.press(button);

        await waitFor(() => expect(getByTestId('error_text')).toBeTruthy());
        expect(getByText('URL must start with http:// or https://')).toBeTruthy();
    });

    test('should not show error when valid URL is entered', async () => {
        const {getByTestId, getByText, queryByTestId} = renderWithReduxIntl(
            <SelectServer {...baseProps}/>,
        );

        const urlInput = getByTestId('server_url_input');
        fireEvent.changeText(urlInput, 'http://localhost:8065');

        const button = getByText('Connect');
        fireEvent.press(button);

        expect(queryByTestId('error_text')).toBeNull();
        await waitFor(() => expect(getByText('Connecting...')).toBeTruthy());
    });
```